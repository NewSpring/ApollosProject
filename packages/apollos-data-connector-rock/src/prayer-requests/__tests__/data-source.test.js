import ApollosConfig from '@apollosproject/config';
import PrayerRequests from '../data-source';
import { buildGetMock } from '../../test-utils';

ApollosConfig.loadJs({
  ROCK: {
    API_URL: 'https://beta-rock.newspring.cc/api',
    API_TOKEN: 'some-rock-token',
    IMAGE_URL: 'https://beta-rock.newspring.cc/GetImage.ashx',
  },
});

describe('PrayerRequests', () => {
  it('constructs', () => {
    expect(new PrayerRequests()).toBeTruthy();
  });

  it('gets all prayer requests', () => {
    const dataSource = new PrayerRequests();
    dataSource.get = buildGetMock({ Id: 51 }, dataSource);
    const result = dataSource.getPrayerRequests();
    expect(result).resolves.toMatchSnapshot();
    expect(dataSource.get.mock.calls).toMatchSnapshot();
  });
});
