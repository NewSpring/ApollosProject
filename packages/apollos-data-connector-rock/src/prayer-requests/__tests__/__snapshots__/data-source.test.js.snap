// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrayerRequests gets all prayer requests 1`] = `
Array [
  Array [
    "PrayerRequests",
    Object {},
    Object {},
  ],
]
`;

exports[`PrayerRequests gets all prayer requests 2`] = `
Object {
  "id": 51,
}
`;
